plugins/CoreVisualizations/Visualizations/EvolutionPeriodSelector.php

```diff
@@ -30,15 +31,14 @@ public function getNumDaysDifference(Date $date1, Date $date2): int
         return (int) round($days);
     }
 
-    public function setSparklineDatePeriods($params, Period $originalPeriod, $comparisonPeriods): array
+    public function setDatePeriods($params, Period $originalPeriod, $comparisonPeriods, $isComparing): array
     {
-        $isComparingDates = !empty($comparisonPeriods);
         $highestPeriodInCommon = $this->getHighestPeriodInCommon($originalPeriod, $comparisonPeriods);
```

Comment:
This one always seems to return the highest period of the shortest comparison periods.
So if someone would e.g. compare `2022-08-09;2022-08-10` with `2016-08-09;2022-08-10`, it would try to use the day period for both. This might still lead to a problem, as the very long second period would be fetched as days.

Not needed to improve that in this PR as well. But was just wondering if we should try to improve that in some way.
We could e.g. restrict that compared periods are more than twice longer than the shortest compared period or something similar.

Reply:
Good point. Indeed it's not ideal. Put it in there for now on purpose so at least both use the same period and it's comparable. With some of the other changes coming it'll be at least a lot faster and not sure if it's done a lot in practice as it wouldn't make all that much sense to compare these periods but can always happen when they select wrong year for example by accident.